crates/rust-analyzer/src/config.rs

```diff
@@ -383,6 +383,14 @@ impl Config {
                     }
                 }
             }
+
+            if let Some(code_action) = &doc_caps.code_action {
+                if let Some(resolve_support) = &code_action.resolve_support {
+                    if resolve_support.properties.iter().any(|it| it == "edit") {
+                        self.client_caps.code_action_resolve = true;
```

Comment:
It's a weird one but we probably need to check for `data_support` too (though I argue why have resolve support without data).